plugins/modules/gem.py

```diff
@@ -334,9 +334,21 @@ def main():
             changed = True
     elif module.params['state'] == 'absent':
         if exists(module):
-            uninstall(module)
-            changed = True
-
+            command_output = uninstall(module)
+            if command_output is not None and exists(module):
```

Comment:
I feel to urge to point out that the return value `None` is being used to determine whether the module is running in check mode or not. This is not a good way to achieve that. In another PR, we should:
* remove the check mode tests from inside the `install`/`uninstall` functions, on the grounds of separation-of-concerns: the function should be concerned about running an external command, rather than whether the module is in check_mode
* here (for uninstall, and in the corresponding spot for install), test if the module is in check mode and act accordingly. Only call the `install`/`uninstall` functions if NOT in check mode.